chalice/package.py

```diff
@@ -30,6 +34,8 @@ def create_app_packager(
     if package_format == 'cloudformation':
         build_stage = create_build_stage(
             osutils, ui, CFNSwaggerGenerator())
+        if merge_template and is_yml_template(merge_template):
```

Comment:
I think it's worth extracting out the serializer to a separate class.  That way to don't have to add conditional `is_yml_template()` logic in all the places we need to branch based on json vs. yaml.  Then we could just pass in the serializer to the template generator.